dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/StaticLinkageChecker.java

```diff
@@ -353,4 +336,25 @@ private boolean validateMethodReference(MethodSymbolReference methodReference) {
         || validateMethodReferenceByClassLoader(methodReference);
   }
 
+  /**
+   * Returns the target class and its superclasses in order (with {@link Object} last). If any can't
+   * be found, the list stops with the previous one.
+   */
+  private Iterable<JavaClass> getClassAndSuperClasses(String targetClassName) {
+    try {
+      return SUPERCLASSES.breadthFirst(classDumper.loadJavaClass(targetClassName));
+    } catch (ClassNotFoundException ex) {
+      return ImmutableList.of();
+    }
+  }
+
+  private static final Traverser<JavaClass> SUPERCLASSES =
+      Traverser.forTree(
+          javaClass -> {
+            try {
+              return ImmutableSet.of(javaClass.getSuperClass());
```

Comment:
java.lang.Object's super class is null in BCEL API. I think that's why it raises NullPointerException.

Reply:
Yes.